src/main/kotlin/net/irisshaders/lilybot/commands/Report.kt

```diff
@@ -37,84 +43,158 @@ class Report : Extension() {
 
             action {
                 val actionLog = guild?.getChannel(MESSAGE_LOGS) as GuildMessageChannelBehavior
+                val reportedMessage = event.interaction.getTarget()
+                val messageAuthor = reportedMessage.getAuthorAsMember()
 
                 respond {
                     content = "Message reported to staff"
                 }
-
-                actionLog.createMessage {
-                    content = "<@&${MODERATORS.value}>"
-                }
-
-                actionLog.createEmbed {
-                    color = DISCORD_RED
-                    title = "Message reported!"
-                    description = event.interaction.getChannel().mention
-
-                    field {
-                        name  = "Messaged Content:"
-                        value = event.interaction.getTarget().content
-                        inline = true
-                    }
-                    field {
-                        name = "Message Link:"
-                        value = event.interaction.getTarget().getJumpUrl()
-                        inline = false
-                    }
-                    footer {
-                        text = "Requested by " + user.asUser().tag
-                        icon = user.asUser().avatar?.url
-                    }
-                    timestamp = Clock.System.now()
-                }
+                createReport(user, actionLog, messageAuthor, reportedMessage)
             }
 
         }
+        // Fuck this thing ngl
         ephemeralSlashCommand(::ManualReportArgs) {
             name = "manual-report"
             description = "Manually report a message"
-            locking = true // To prevent the command from being run more than once concurrently
 
             action {
                 val actionLog = guild?.getChannel(MESSAGE_LOGS) as GuildMessageChannelBehavior
-                val messageID = Snowflake(arguments.message.split("/")[6])
-                val messageChannelID = Snowflake(arguments.message.split("/")[5])
-                val messageChannel = (guild?.getChannel(messageChannelID) as MessageChannel).getMessage(messageID).content
+                val channel = (guild?.getChannel(Snowflake(arguments.message.split("/")[5])) as MessageChannel)
+                val reportedMessage = channel.getMessage(Snowflake(arguments.message.split("/")[6]))
+                val messageAuthor = reportedMessage.getAuthorAsMember()
 
                 respond {
                     content = "Message reported to staff"
                 }
+                createReport(user, actionLog, messageAuthor, reportedMessage)
+            }
+        }
+    }
 
-                actionLog.createMessage {
-                    content = "<@&${MODERATORS.value}>"
-                }
+    inner class ManualReportArgs : Arguments() {
```

Comment:
I would rather the inner class be the last thing in this file please

Reply:
Will do